### PR TITLE
[TC Automation] PTX-21440 Add test case VolAttachFAPxRestart

### PR DESCRIPTION
**Pull request by spai-px**
**What this PR does / why we need it**:
Adds test case VolAttachFAPxRestart 

**Which issue(s) this PR fixes** (optional)
Closes # PTX-21440 

**Special notes for your reviewer**:
[Jenkins run](https://jenkins.pwx.dev.purestorage.com/job/Users/job/krishna/job/tp-pxe-byoc/2680/console)
[Aetos link](https://aetos.pwx.purestorage.com/resultSet/testSetID/621355 )

### CODE REVIEW

File: tests/basic/pure_test.go
Comment (spai-px) [reply tests/basic/pure_test.go:2406]: we need this to get secrets not present in pure.json

File: pkg/pureutils/pure_types.go
Comment (spai-px) [reply pkg/pureutils/pure_types.go:81]: So the test case is to attach a volume from FA which is not present in pure.json so I have created another file named non-pure.json which should contain the secrets not present in pure.json.

File: pkg/pureutils/pure.go
Comment (spai-px) [reply pkg/pureutils/pure.go:160]: Done.

File: tests/basic/pure_test.go
Comment (spai-px) [reply tests/basic/pure_test.go:2644]: Done.

File: tests/basic/pure_test.go
Comment (vprabhakar-px) [reply tests/basic/pure_test.go:2406]: why to get secrets , instead why not pass details from torpedo externally

File: pkg/pureutils/pure_types.go
Comment (spai-px) [reply pkg/pureutils/pure_types.go]: Changed it to take it from env variables.

File: tests/basic/pure_test.go
Comment (spai-px) [reply tests/basic/pure_test.go]: Yeah done.

File: tests/basic/pure_test.go
Comment (vprabhakar-px) [inline tests/basic/pure_test.go:2441]: pls update the comment

File: tests/basic/pure_test.go
Comment (vprabhakar-px) [inline tests/basic/pure_test.go:2446]: instead of hardcoding pick random Volume ranges from few 1GB till 100Gb

File: tests/basic/pure_test.go
Comment (vprabhakar-px) [inline tests/basic/pure_test.go:2454]: Validate if IQN is already present on some hosts

File: tests/basic/pure_test.go
Comment (vprabhakar-px) [inline tests/basic/pure_test.go:2550]: Split the function to Login to one specific controller and use the function to access and login from multiple network interfaces

**Review by vprabhakar-px — APPROVED** [review]
some changes Needed , LGTM after updating all the comments

File: tests/basic/pure_test.go
Comment (spai-px) [reply tests/basic/pure_test.go]: Done.

File: tests/basic/pure_test.go
Comment (spai-px) [reply tests/basic/pure_test.go]: Done. If Iqn is present taking the host which is already present.

File: tests/basic/pure_test.go
Comment (spai-px) [reply tests/basic/pure_test.go]: Done.

File: tests/basic/pure_test.go
Comment (spai-px) [reply tests/basic/pure_test.go]: Done.

**Review by dbhatnagar-px — APPROVED** [review]
Lgtm

**Pull request merged**
